### PR TITLE
No-results and Error empty state

### PR DESCRIPTION
**Pull request by shiranmeir**
New No-resutls empty state icon
New empty-states to Courses and Users tables

Added empty-error state with new icon

### CODE REVIEW

File: src/components/common/WMEmpty/SearchEmptyState.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import Icon, { IconType } from '../Icon';
+
+import classes from './style.module.scss';
+
+export default function CoursesEmptyState(): ReactElement {
+  return (
+    <div className={classes['search-empty-state']}>
+      <Icon className={classes['icon']} type={IconType.EmptySearch} />
+      <h1>No Results</h1>
+      <p>Please try different search</p>
```
Comment (dvira) [inline src/components/common/WMEmpty/SearchEmptyState.tsx:12]: Was this wording derived from the design? It seems a bit substandard. Please consider rephrasing it.

File: src/images/icons/empty-error.svg [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+<path d="M224.985 106.912c-4.343-9.464.845-13.395 14.21-11.373a1 1 0 0 1-.299 1.978l-.749-.11c-11.43-1.594-14.789 1.164-11.345 8.67 2.954 6.438 15.842 16.18 31.932 23.644 19.01 8.818 39.7 13.46 58.178 12.026l.838-.122c13.42-2.059 14.238-6.814 2.735-15.767a1 1 0 1 1 1.229-1.578c13.098 10.194 11.49 17.208-4.586 19.455-18.914 1.468-39.935-3.247-59.236-12.2-16.486-7.647-29.702-17.638-32.907-24.623z" fill="#C6C6C6" fill-rule="nonzero"/>
+<path d="M197.5 153a4 4 0 1 1 0 8 4 4 0 0 1 0-8zm213-9a4 4 0 1 1 0 8 4 4 0 0 1 0-8zm-319-11a4 4 0 1 1 0 8 4 4 0 0 1 0-8zm-81-19a4 4 0 1 1 0 8 4 4 0 0 1 0-8zm199-44a4 4 0 1 1 0 8 4 4 0 0 1 0-8zm-45-29a4 4 0 1 1 0 8 4 4 0 0 1 0-8zM44 161a2.5 2.5 0 1 1 0 5 2.5 2.5 0 0 1 0-5zm142-43a2.5 2.5 0 1 1 0 5 2.5 2.5 0 0 1 0-5zm-47-2a2.5 2.5 0 1 1 0 5 2.5 2.5 0 0 1 0-5zm235-7a2.5 2.5 0 1 1 0 5 2.5 2.5 0 0 1 0-5zM237 34a2.5 2.5 0 1 1 0 5 2.5 2.5 0 0 1 0-5zM60 9a2.5 2.5 0 1 1 0 5 2.5 2.5 0 0 1 0-5zm52-9a2.5 2.5 0 1 1 0 5 2.5 2.5 0 0 1 0-5z" fill="#D8D8D8"/>
+</g>
+</svg>
```
Comment (dvira) [inline src/images/icons/empty-error.svg:23]: Add an empty line at the end of this file.

File: src/images/icons/empty-search.svg [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+</g>
+<g transform="translate(0 3)">
+<path fill="#D7D7D7" transform="translate(129 39)" d="M70.21 67.145c.47.393.707.904.707 1.532 0 .629-.216 1.16-.649 1.591a2.163 2.163 0 0 1-1.59.649c-.63 0-1.14-.236-1.533-.707L48.17 53.59c-2.514 1.965-5.343 3.497-8.486 4.597a29.054 29.054 0 0 1-9.664 1.65c-5.422 0-10.431-1.336-15.027-4.007-4.597-2.672-8.231-6.306-10.902-10.902C1.419 40.332.083 35.343.083 29.96c0-5.383 1.336-10.372 4.008-14.968 2.671-4.597 6.305-8.231 10.902-10.902C19.589 1.419 24.578.083 29.96.083c5.382 0 10.371 1.336 14.968 4.008 4.596 2.671 8.23 6.305 10.902 10.902 2.671 4.596 4.007 9.605 4.007 15.027 0 3.928-.727 7.66-2.18 11.196a29.524 29.524 0 0 1-6.188 9.429l18.74 16.5zM4.562 30.02c0 4.557 1.14 8.78 3.418 12.67 2.279 3.889 5.362 6.973 9.252 9.251 3.89 2.279 8.132 3.418 12.729 3.418 4.596 0 8.84-1.139 12.728-3.418 3.89-2.278 6.974-5.362 9.252-9.252 2.279-3.889 3.418-8.132 3.418-12.728 0-4.597-1.139-8.84-3.418-12.73-2.278-3.889-5.362-6.972-9.252-9.251C38.8 5.7 34.557 4.562 29.961 4.562c-4.597 0-8.84 1.14-12.73 3.418-3.889 2.279-6.972 5.362-9.251 9.252-2.279 3.89-3.418 8.152-3.418 12.788z" />
+</g>
+</g>
+</svg>
```
Comment (dvira) [inline src/images/icons/empty-search.svg:46]: Add an empty line at the end of this file.

File: src/components/Screen/ErrorScreen/index.tsx [hunk 4]
```diff
@@ -1,15 +1,19 @@
 import React, { ReactElement, useState } from 'react';
 
+import Icon, { IconType } from '../../common/Icon';
+
 import classes from './style.module.scss';
 
 export default function ErrorScreen({ error }: { error?: string }): ReactElement {
   const [errorMessage, setErrorMessage] = useState(
-    error && Boolean(error) ? error : 'Something went wrong',
+    error && Boolean(error) ? error : 'Something is wrong',
```
[the remark this answers is not in the document]
Comment (shiranmeir) [reply src/components/Screen/ErrorScreen/index.tsx:9]: Ok, changed it

File: src/components/Screen/ErrorScreen/index.tsx [hunk 5]
```diff
@@ -1,15 +1,19 @@
...
   );
 
   return (
     <div className={classes['error-screen']}>
-      <p>{errorMessage}</p>
+      <Icon type={IconType.EmptyStateError} />
+      <h1>{errorMessage}</h1>
+      <p>Please try again</p>
```
[the remark this answers is not in the document]
Comment (shiranmeir) [reply src/components/Screen/ErrorScreen/index.tsx:16]: Added

File: src/components/Screen/UsersScreen/index.tsx [hunk 6]
```diff
@@ -16,6 +17,9 @@ import WMCard from '../../common/WMCard';
 import WMTable from '../../common/WMTable';
 import ScreenHeader from '../../common/ScreenHeader';
 import ControlsWrapper from '../../common/ControlsWrapper';
+import SearchEmptyState from '../../common/WMEmpty/SearchEmptyState';
+import DataEmptyState from '../../common/WMEmpty/DataEmptyState';
```
[the remark this answers is not in the document]
Comment (shiranmeir) [reply src/components/Screen/UsersScreen/index.tsx:21]: Ok did it

File: src/components/common/Icon/icon.interface.ts [hunk 7]
```diff
@@ -24,6 +24,8 @@ export enum IconType {
   QuizSettings = 'quiz-settings',
   EmptyCourse = 'empty-course',
   EmptyLesson = 'empty-lesson',
+  EmptySearch = 'empty-search',
+  EmptyStateError = 'empty-error',
```
[the remark this answers is not in the document]
Comment (shiranmeir) [reply src/components/common/Icon/icon.interface.ts:28]: Right, Ok

File: src/components/common/Icon/iconsSVG.tsx [hunk 8]
```diff
@@ -60,6 +62,8 @@ export const IconSVG = {
   [IconType.QuizSettings]: QuizSettingsIcon,
   [IconType.EmptyCourse]: EmptyCourseIcon,
   [IconType.EmptyLesson]: EmptyLessonIcon,
+  [IconType.EmptySearch]: EmptySearchIcon,
+  [IconType.EmptyStateError]: EmptyStateErrorIcon,
```
[the remark this answers is not in the document]
Comment (shiranmeir) [reply src/components/common/Icon/iconsSVG.tsx:66]: Fixed it

File: src/components/common/WMEmpty/DataEmptyState.tsx [hunk 9]
```diff
@@ -0,0 +1,12 @@
+import React, { ReactElement } from 'react';
+import { Empty } from 'antd';
+
+import classes from './style.module.scss';
+
+export default function CoursesEmptyState(): ReactElement {
```
[the remark this answers is not in the document]
Comment (shiranmeir) [reply src/components/common/WMEmpty/DataEmptyState.tsx:6]: Tnx

File: src/components/common/WMEmpty/SearchEmptyState.tsx [hunk 10]
```diff
@@ -0,0 +1,15 @@
+import React, { ReactElement } from 'react';
+
+import Icon, { IconType } from '../Icon';
+
+import classes from './style.module.scss';
+
+export default function CoursesEmptyState(): ReactElement {
```
[the remark this answers is not in the document]
Comment (shiranmeir) [reply src/components/common/WMEmpty/SearchEmptyState.tsx:7]: Tnx

File: src/images/icons/empty-error.svg (hunk 2 above)
Comment (shiranmeir) [reply src/images/icons/empty-error.svg:23]: Ok

File: src/images/icons/empty-search.svg (hunk 3 above)
Comment (shiranmeir) [reply src/images/icons/empty-search.svg:46]: Ok

File: src/components/common/WMEmpty/SearchEmptyState.tsx (hunk 1 above)
Comment (shiranmeir) [reply src/components/common/WMEmpty/SearchEmptyState.tsx]: Changed it

File: src/components/common/WMEmpty/index.tsx [hunk 11]
```diff
@@ -1,10 +1,16 @@
...
 import { EmptyProps } from 'antd/lib/empty';
 import cc from 'classcat';
 
+import DataEmptyState from './DataEmptyState';
+import SearchEmptyState from './SearchEmptyState';
+
 import classes from './style.module.scss';
 
+export { DataEmptyState };
+export { SearchEmptyState };
```
Comment (dvira) [inline src/components/common/WMEmpty/index.tsx:12]: No need for 2 exports, one for both will do.

File: src/components/Screen/UsersScreen/index.tsx [hunk 12]
```diff
@@ -16,11 +17,13 @@ import WMCard from '../../common/WMCard';
 import WMTable from '../../common/WMTable';
 import ScreenHeader from '../../common/ScreenHeader';
 import ControlsWrapper from '../../common/ControlsWrapper';
+
```
Comment (dvira) [inline src/components/Screen/UsersScreen/index.tsx:20]: Why the added space?

File: src/components/common/WMEmpty/index.tsx (hunk 11 above)
Comment (shiranmeir) [reply src/components/common/WMEmpty/index.tsx:12]: Ok

File: src/components/Screen/UsersScreen/index.tsx (hunk 12 above)
Comment (shiranmeir) [reply src/components/Screen/UsersScreen/index.tsx:20]: Removed

**Pull request merged**